picard/ui/mainwindow.py

```diff
@@ -1440,7 +1448,9 @@ def update_selection(self, objects=None, new_selection=True, drop_album_caches=F
                                                history=None)
         elif coverart_visible and new_selection:
             # Create a temporary file list which allows changing cover art for all selected files
-            files = self.tagger.get_files_from_objects(objects)
+            files = list(iter_files_from_objects(objects))
+            # TODO: it should be possible to pass the generator directly to FileList(),
+            # but underlying code has to be checked
```

Comment:
Actually no, at least not in general. The passed file list is passed on to `FileListItem`, and that just sets the passed value. And some entity objects assume this is a list and use `len()` or index access on files.

It might for some reason work in this specific case where we have this temporary `FileList` currently only for cover art updating. But if you pass on an iterator you also pass on state, and I wouldn't keep this iterator around for storing the actual list of files attached to an object for longer. E.g. `iter_files_from_objects(objects)` depends on iterating over the list of files of each object, and that might change. So the files we currently think we pass on to FileList might not be the ones that get later used when this iterator is finally evaluated.

Reply:
I removed the TODO comment and rebased